File: SchoolWebApp-API/Project.Core/Entities/Students/StudentSubject.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.CBE.Exams;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Students
{
    public class StudentSubject : Base
    {
        public int StudentClassId { get; set; }
        public StudentClass? StudentClass { get; set; }
        public int SubjectId { get; set; }
        public Subject? Subject { get; set; }
        public string? Description { get; set; }
        public List<ExamResult> ExamResults { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/students/models/student-subject.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import {StudentClass} from './student-class';
import {Subject} from '@/academics/models/subject';

export class StudentSubject extends ResourceModel<StudentSubject> {
    public description?: string;

    public studentClassId?: number;
    public studentClass?: StudentClass;

    public subjectId?: number;
    public subject?: Subject;

    constructor(model?: Partial<StudentSubject>) {
        super(model);
    }
}
